apps/arweave/src/ar_merkle.erl

```diff
@@ -190,6 +196,112 @@ validate_path_strict_data_split(_ID, _Dest, _LeftBound, _RightBound, _Path, _Pat
 		_DataSize) ->
 	false.
 
+%% @doc Validate the given merkle path where any subtrees may have 0-based offset.
+validate_path2(ID, Dest, RightBound, _Path) when RightBound =< 0 ->
+	?LOG_ERROR([{event, validate_path2_called_with_not_positive_right_bound},
+			{root, ar_util:encode(ID)}, {dest, Dest}, {right_bound, RightBound}]),
+	throw(invalid_right_bound);
+validate_path2(ID, Dest, RightBound, Path) when Dest >= RightBound ->
+	validate_path2(ID, RightBound - 1, RightBound, Path);
+validate_path2(ID, Dest, RightBound, Path) when Dest < 0 ->
+	validate_path2(ID, 0, RightBound, Path);
+validate_path2(ID, Dest, RightBound, Path) ->
+	validate_path2(ID, Dest, 0, RightBound, Path, byte_size(Path), RightBound, 0).
+
+validate_path2(ID, _Dest, LeftBound, RightBound,
+		<< Data:?HASH_SIZE/binary, EndOffset:(?NOTE_SIZE*8) >>, PathSize, DataSize,
+		LeftBoundShift) ->
+	case EndOffset - LeftBound > ?DATA_CHUNK_SIZE
+			orelse RightBound - LeftBound > ?DATA_CHUNK_SIZE of
+		true ->
+			io:format("Bounds: E=~B L=~B R=~B~n", [EndOffset, LeftBound, RightBound]), % TODO
+			false;
+		false ->
+			ChunkSize = EndOffset - LeftBound,
+			ValidateSplit =
```

Comment:
Does this value store whether or not the split is valid? What do you think about naming it `IsSplitValid` or something like that?